### DISCUSSION TITLE
Add beatmap slider tickrate to Difficulty Adjust mod settings

### DISCUSSION THREAD

**Discussion by apollo-dw** [ppy/osu#16552] [Ideas]
sounds like a fun idea to me, would effectively change the difficulty of following sliders in a map. down to try adding this if there's support, it's definitely something that i would use

**Comment by bdach** [comment]
maybe a touch specific/technical? but sure, don't really see a huge reason not to add this myself

**Reply by bdach** [reply]
if this happens though then stack leniency should probably also happen as well (i can see that one being more useful/popular than tick rate actually)

**Reply by apollo-dw** [reply]
stack leniency is a good idea too, and i could safely say that at a high level there will be select players who will be very interested in increasing tick rate on maps :P

**Comment by peppy** [comment]
Weakly against this. Seems like one of those options that will be used once, ever.

**Reply by mk56-spn** [reply]
youd be surprised. an example would be this play by trail mix

https://www.youtube.com/watch?v=AtwkzESVYY0&ab_channel=TrailMixOsu%21

he gets no PP since theres like 3 slider ticks even though hes playing the slider properly. if this was a thing he would be able to adjust the tickrate and more fairly represent his skill

**Reply by smoogipoo** [reply]
I'm also on the line about this, but let me give a reason for why this would not be a good idea on notch hell specifically.

A slider tick option affects every slider in the map, so while it would be great to see those large fast kiai sliders be followed more accurately and give more PP for those willing to follow them correctly, every other slider which players don't even really think about, will suck to play. So I'm tending more towards players will be finding this option annoying rather than not, though I'd love to be proven wrong on this. I'm the type of player that tries to follow these types of sliders accurately, but haven't gotten the chance to try it out myself yet: https://github.com/ppy/osu/discussions/16552#discussioncomment-2040800

**Reply by mk56-spn** [reply]
well the thing is smoogi, on normal sliders if you are following well enough to hit the slider ends, which is waht most platers are doing then hitting whats inbetween is a given bar some obnoxious shape, so idk , i understand your qualms but i dont share them per say

**Comment by smoogipoo** [comment]
Those curious can test out how it'd behave by applying the following patch:
```diff
diff --git a/osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs b/osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs
index eefa1531c4..bc9686ee49 100644
--- a/osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs
+++ b/osu.Game/Rulesets/Mods/ModDifficultyAdjust.cs
@@ -52,6 +52,15 @@ public abstract class ModDifficultyAdjust : Mod, IApplicableToDifficulty
             ReadCurrentFromDifficulty = diff => diff.OverallDifficulty,
         };

+        [SettingSource("Tick Rate", "", LAST_SETTING_ORDER + 1, SettingControlType = typeof(DifficultyAdjustSettingsControl))]
+        public DifficultyBindable TickRate { get; } = new DifficultyBindable
+        {
+            Precision = 0.1f,
+            MinValue = 0.1f,
+            MaxValue = 10f,
+            ReadCurrentFromDifficulty = _ => 1
+        };
+
         [SettingSource("Extended Limits", "Adjust difficulty beyond sane limits.")]
         public BindableBool ExtendedLimits { get; } = new BindableBool();

@@ -93,6 +102,7 @@ protected virtual void ApplySettings(BeatmapDifficulty difficulty)
         {
             if (DrainRate.Value != null) difficulty.DrainRate = DrainRate.Value.Value;
             if (OverallDifficulty.Value != null) difficulty.OverallDifficulty = OverallDifficulty.Value.Value;
+            if (TickRate.Value != null) difficulty.SliderTickRate *= TickRate.Value.Value;
         }
     }
 }
```

**Reply by mk56-spn** [reply]
@smoogipoo two issues ive noticed, it seems to mess with the slider audiom (tested on tickrate 10, on the famous map scarlet rose)
and the hp drain is absolutely barbaric for missing a slider, those two can be adressed thought (making audio treat the map like its original tickrate maybe?)
